### ISSUE TITLE
Publication to OBIS

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#3484]
In a meeting with GBIF North American node manager today, she asked if there are any Arctos collections interested in publishing to the Ocean Biodiversity Information System [OBIS](https://obis.org/). I'm not sure how this would be organized, but I hope it could be done via the VertNet ipt. If anyone is interested, let me know and I'll try to set up a discussion about it.

@sharpphyl @acdoll @lin-fred

**Comment by dustymc** [comment]
https://obis.org/manual/contribute/ says they speak DWC so I don't think anything would need "published," just tell them where the DWC files are (http://ipt.vertnet.org:8080/ipt/).

**Comment by campmlc** [comment]
So these folks harvest from WoRMS?
They also have marine mammals and birds.

**Comment by Jegelewicz** [comment]
Yes OBIS uses WoRMS for taxonomy!

**Comment by lin-fred** [comment]
As far as what they state about data quality, what about the freshwater specimens that are mixed in with our ocean specimens? Would it know not to take them?

**Comment by acdoll** [comment]
> what about the freshwater specimens

We have the same issue with terrestrial species sprinkled throughout our data.

**Comment by campmlc** [comment]
There appear to be some terrestrial species in their data. I searched on
Cestoda and a significant number were flagged as terrestrial.

**Comment by sharpphyl** [comment]
It looks like they may have the ability to filter out non-marine records.  I see no issue with going forward if our records meet their criteria.

**Comment by albenson-usgs** [comment]
Hi! I'm the GBIF-US Node manager. I'm also the OBIS-USA node manager. A couple points regarding publishing to OBIS.
1. Freshwater specimens need to be filtered out in publishing to OBIS. OBIS can "filter" them in a sense but they will show up as data quality flags ([example](https://obis.org/dataset/b04ae33b-2f2a-4d06-844a-4f2040993fcc) of a few we missed for the UF Invertebrate Collection. I have worked with the [UF Invertebrate Collection](https://www1.usgs.gov/obis-usa/ipt/resource?r=flmnh_iz) and the [KUBI Ichthyology Collection](https://www1.usgs.gov/obis-usa/ipt/resource?r=kubi_ichthyologycollection_marine) previously to do this. I usually use a combination of environment flags from WoRMS and geospatial filters to filter out records on land.
2. The data will need to be published via the OBIS-USA IPT because 1) it's registered with OBIS and VertNet's is not and more importantly 2) this is how we'll share the marine specimens separate from the freshwater/terrestiral specimens to meet 1 above
3. OBIS does have a few extra requirements that GBIF does not have. Required Darwin Core terms for OBIS:
eventDate 
decimalLatitude
decimalLongitude
scientificName 
scientificNameID (must be the LSID from WoRMS)
occurrenceStatus 
basisOfRecord 
occurrenceID 
datasetID 

Happy to help as needed! Just let me know.

**Comment by sharpphyl** [comment]
@acdoll  Would you want to contribute marine birds and mammals as well as marine invertebrates?
Our Marine Invertebrate taxonomy should be in good condition as it uses valid WoRMS taxa.  

@dustymc  Does most of this happen automatically or would we (or you) have to do a lot of preparation to meet the OBIS standards?

@albenson-usgs  It appears that only records with marine lat/longs are published.  Does OBIS eliminate what isn't useable or do we need to do that?

**Comment by albenson-usgs** [comment]
@sharpphyl publishing to OBIS would not happen automatically and it's preferable for data providers to work with me to get missing elements added and subset out records that are not appropriate. The subset would need to be shared via the [OBIS-USA IPT](https://www1.usgs.gov/obis-usa/ipt/) as the VertNet IPT is not registered with OBIS. Does that help?

**Comment by Jegelewicz** [comment]
So, this definitely looks like extra work to me and I am not sure that we have the resources right now to make it happen unless @dustymc has magic. We would need:

 - [ ] Create a DarwinCore file that includes ONLY marine records (by taxon or by locality or both?) AND only includes collections that wish to participate.
 - [ ] Share that DwC file to the OBIS-USA IPT
 - [ ] Refresh - how often?

It is the first step that is problematic and seems like a lot of work to me. @albenson-usgs this is a prime example of why collections will fail to participate as we are being required to provide our data in some new form/subset. We already do this for GBIF/iDigBio, GGBN, GloBI. In addition, our data is all already publicly available in far more detail than any of these aggregators can publish. I'm not saying we won't eventually do this, but we really don't have resources to create new facets of our data for every possible use case and I am certain that all of the other CMS feel the same. Sorry if this sounds like whining - I just really think it is something that the larger community of users should be considering.

**Comment by sharpphyl** [comment]
I agree with @Jegelewicz  that it sounds like work that right now we don't have the resources to do.  If multiple Arctos collections wanted to publish in OBIS, then it would make sense to expend the resources necessary.  Perhaps a topic for the AWG to consider at some point.  For just our relatively small collection (compared to the UF invertebrate collection), the time and effort would be greater than the benefit.  We do already have significant visibility with GBIF and InvertEBase, etc.  But it's good to know what would be required if we wanted to participate.

**Comment by albenson-usgs** [comment]
@Jegelewicz and @sharpphyl I do understand. It would not fall completely on you all to do the work necessary. I do have code (in R) already for the UF Invertebrate and KUBI collections that can be reused. However, I need someone to work with to ask questions of that will be responsive (e.g. "these are the species to be subsetted out- does that look right to you?"). The one benefit I could see from doing this via Arctos as opposed to going collection by collection is that I'm assuming the data are all formatted pretty much the same so we may be able to develop a workflow wherein any datasets that have a WoRMS identifier are identified. Any records from that dataset without a WoRMS identifier are subsetted out from the dataset then (maybe) I can have an "Arctos script" that those datasets with just those records and adds any missing columns/checks for points on land. If the data coming into that "Arctos script" are consistent in where the information is located and what's missing it should be an easy thing to re-run the script across multiple datasets. Does that make sense? But I need someone to ask questions when the script finds things that seem anomalous to me.

**Comment by Jegelewicz** [comment]
@albenson-usgs we talked about this yesterday in our taxonomy committee meeting. We are up for you testing, but it is difficult to make promises about how responsive we can be or who is best to answer questions. Anyway, if you are willing to deal with the possibility of waiting around for a response, we think we can go ahead. So, first question:

> I do have code (in R) already for the UF Invertebrate and KUBI collections that can be reused. 

But how will you access our data? We would prefer that you do that from the VertNet IPT. Possible? Then you could ask questions as they arise?

**Comment by albenson-usgs** [comment]
Yes, that's possible. I have some other data processing that I'm working through so I'm not exactly sure when I'll get to this but I will add it to my to do list :-) Can you point me at one particular dataset that would be good to start with? I only see a link to the entire VertNet IPT in this thread. Also I want to let @dbloom know that this is happening just for awareness.

**Comment by Jegelewicz** [comment]
The DMNS:Inv dataset would be a good test case. It is Arctos collection number 74. Let me know if that doesn't get you what you need!

**Comment by albenson-usgs** [comment]
It's this one: http://ipt.vertnet.org:8080/ipt/resource?r=dmns_inv, right? It's showing up as red in the IPT which I've never seen before. Does anyone know why?

**Comment by Jegelewicz** [comment]
@dustymc do you know what the above means?

**Comment by dustymc** [comment]
Nope.

**Comment by albenson-usgs** [comment]
I'll check with Dave.

**Comment by dbloom** [comment]
@albenson-usgs per my Slack message to you: this happens when resources on an automated publishing schedule fail to publish properly. I noticed that there were two datasets from Arctos that failed in the last publication event a few days ago. Both are direct dbase connections, so after checking on the IPT side I notified Dusty who remapped and reset Arctos for me (something that take several hours to complete - thank you Dusty) and now both resources have been republished and are working normally.

**Comment by Jegelewicz** [comment]
@albenson-usgs any progress here - anything we need to do?

**Comment by albenson-usgs** [comment]
Thanks for the nudge! Let me see if this is one I can get a student I'm working with to help me with. Too many irons in the fire.

**Comment by Jegelewicz** [comment]
No worries - I was just going through old issues trying to pare down the pile!

**Comment by albenson-usgs** [comment]
Apologies for the reply during the break but I finally had some quiet time to make progress on this. Attached are three files from the DMNS Inverts collection that need a collection manager to review them. What I have done (with assistance from a student I'm working with) is create a unique list of scientific names then match those names to the World Register of Marine Species using [this code](https://github.com/albenson-usgs/BCB_DataWranglingAndAnalyses/blob/master/OBIS_DwC_Alignment/DMNS_Inverts_OBISMapping.R). Once the matching was done I created three subsets 1) scientific names that were not able to be matched (note that this includes scientific names with cf., aff., etc in them because the code can't match those and it might be worth considering parsing that information into `identificationQualifier`) 2) scientific names that WoRMS suggests an updated accepted name, 3) scientific names that found a match in WoRMS but had environmental flags that indicated they are not marine and are not brackish species. 
[DMNS_UniqueTaxa_NoWoRMSMatch.csv](https://github.com/ArctosDB/arctos/files/7793998/DMNS_UniqueTaxa_NoWoRMSMatch.csv)
[DMNS_WoRMS_SuggestsUpdatedAcceptedName.csv](https://github.com/ArctosDB/arctos/files/7793999/DMNS_WoRMS_SuggestsUpdatedAcceptedName.csv)
[DMNS_Inverts_MatchedtoWoRMS_butNotMarine.csv](https://github.com/ArctosDB/arctos/files/7794000/DMNS_Inverts_MatchedtoWoRMS_butNotMarine.csv)

Where to go from here:
1. We can simply remove all the non-matches and non-marine specimens and decide not to do anything with the suggestions for accepted name. This is the most simple but means probably sharing the smallest subset to OBIS. The next step for me here would be to check for occurrences on land.
2. DMNS can review the files and determine if they agree with the designations. Update their collection if they find places that it makes sense to do so. Interact with WoRMS (very responsive group) about places they disagree. Republish their collection via the VertNet IPT and I redo our process. Longer process but results in a better subset for OBIS.

I'm open to other ways forward as well but those are the two that I see.

**Comment by Jegelewicz** [comment]
> scientific names that were not able to be matched (note that this includes scientific names with cf., aff., etc in them because the code can't match those and it might be worth considering parsing that information into identificationQualifier

See https://github.com/ArctosDB/arctos/issues/2172 and https://github.com/ArctosDB/arctos/issues/2680

**Comment by sharpphyl** [comment]
@albenson-usgs  I'm just starting to look at these files and will get back to you on your questions in a few days.  

We work constantly to identify unaccepted names and update them.  I have about 100 that I'm working on right now, so I'll work on your list as well.

I do not know how to separate our non-marine from marine species beyond what you have done.  I assume you would omit those records from your upload to OBIS.

How often would you upload?  Our database grows and improves constantly.  

Thanks for all your efforts to try to make this data of value to OBIS and beyond.

**Comment by albenson-usgs** [comment]
@sharpphyl sounds great and completely understand about the unaccepted names. Thanks for taking a look.

For the non-marine list, it's 1624 names that are in that list. Can you confirm that sounds like about the right number of non-marine species you have? Or if you are able to look over the list and determine these are the right ones to be subsetting out (but I realize that would be more time consuming). Yes I need to remove these for the upload to OBIS.

How often is a good question. If we get all these details hammered out and nothing changes in how the data are formatted then it should be a simple re-run of the script and possibly a recheck of the different subsets to make sure everything looks correct still. I could imagine doing that annually.

**Comment by Jegelewicz** [comment]
@albenson-usgs just letting you know that we update the IPT monthly - no pressure to update OBIS! The extra work on your end makes it understandable that an annual update is the best we can do.

@sharpphyl thanks for working on this!

**Comment by sharpphyl** [comment]
@albenson-usgs We're doing a major update to our current records.  If updates are annual, I'd like to wait a few weeks until we complete this work so the data is more accurate.  Let me know if that's any problem.  

As for non-marine records, I show that we have 2,187 records (1,187 scientific names) in Stylommatophora.  I'll look at the fresh water over the weekend, but your estimate of 1,624 names looks to be "about the right number of non-marine species."

Thanks again for all your effort to get our records into OBIS.

**Comment by sharpphyl** [comment]
@Jegelewicz 

> Seems like step one should be dealing with the cf. aff. and A {string} IDs appropriately in DwC before we spend a bunch of time looking for other problems.

What's being done on this.  Is it anything I need to deal with or is it in the Arctos structure?

@albenson-usgs  Most of the remaining taxon names without a WoRMS match are in my list of invalid taxa to correct over the next few weeks.  Thanks for running these WoRMS reports.  I don't want to delay the upload forever, but since we're already working on many of these, a few weeks delay may be worthwhile.

**Comment by Jegelewicz** [comment]
> What's being done on this. Is it anything I need to deal with or is it in the Arctos structure?

It is Arctos structure - see See https://github.com/ArctosDB/arctos/issues/2172 and https://github.com/ArctosDB/arctos/issues/2680

I have failed miserably at getting anyone to notice or care about how our data looks at the aggregators....

**Comment by sharpphyl** [comment]
@albenson-usgs 
I've finished most of the updates that we discussed above.  All the invalid taxon names have been corrected.  There are still names that you listed as having no WoRMS match, but I think all of them are terrestrial or freshwater species that you would omit from your upload anyway.  

We have only a handful (less than 50) records that use cf. or aff. and it looks like waiting for those records would be a longer delay, so I suggest that we omit them from the first upload.  @Jegelewicz  Do we still need to discuss [#2680](https://github.com/ArctosDB/arctos/issues/2680) in the Taxonomy Committee or is that now on the AWG agenda?

We also recently added "individual count" to all our records so I'd to have you wait to upload the data until after our next upload to VertNet.  

If you see any other issues that need attention, just let me know.  Thanks for your patience and willingness to include our collection data.

@dustymc Can you let us know when the next upload to VertNet is done so that all these changes appear in the data that Abby collects?

**Comment by Jegelewicz** [comment]
>  "individual count" to all our records so I'd to have you wait to upload the data until after our next upload to VertNet.

@sharpphyl this won't populate properly unless @dustymc has changed the IPT mapping and I can't remember if that has happened.

**Comment by Jegelewicz** [comment]
As for #2680 I totally failed at discussing that in either taxonomy or office hours! This is actually on hold as we are waiting on an MOU with Dave so that we can pay him for this kind of work. Somehow I need to remember that this needs doing as soon as we can.

**Comment by sharpphyl** [comment]
> "individual count" to all our records so I'd to have you wait to upload the data until after our next upload to VertNet.

> @sharpphyl this won't populate properly unless @dustymc has changed the IPT mapping and I can't remember if that has happened.

Dusty added it to all our records, but he can confirm if it's also in our IPT mapping.

<img width="460" alt="Screen Shot 2022-02-25 at 9 17 24 AM" src="https://user-images.githubusercontent.com/15368365/155749602-1f116b4a-6fde-4309-9cab-bdb491e37a9a.png">

**Comment by dustymc** [comment]
Nothing about IPT should need to change, and I have no idea when the data are pulled.

**Comment by Jegelewicz** [comment]
> Nothing about IPT should need to change

But I thought that currently we were only sending part lot count as dwc:individualCount for fishes and other stuff was just getting 1 or NULL? Don't we need to now map this attribute to dwc:individualCount to affect any change in @sharpphyl data at GBIF?

**Comment by albenson-usgs** [comment]
After bugging Dave Bloom I was able to find out that the dataset was [republished on March 2 in the IPT](http://ipt.vertnet.org:8080/ipt/resource?r=dmns_inv) but I still don't know if the individualCount issue is resolved? Am I ok to grab the March 2 version of the data?

**Comment by sharpphyl** [comment]
Yes.  I just checked a few records and they all show the accurate Individual Count (which was added 2022-02-21).

**Comment by albenson-usgs** [comment]
@sharpphyl OBIS is piloting a new way of getting datasets that are already shared to GBIF also shared to OBIS directly from the IPT rather than me harvesting the dataset and putting it on the OBIS-USA IPT. It takes identifying OBIS as a Network in the IPT for the dataset. I think I need to work with @dbloom on that. Would this be ok from your perspective. It means that when updates are published to the IPT they will automatically be harvested by OBIS.

**Comment by sharpphyl** [comment]
@acdoll Do you have any concerns?  

It sounds fine to me, Abby.  In fact, this sounds like an easier approach that would result in more complete data.

**Comment by acdoll** [comment]
Nope, that sounds good to me!

**Comment by dbloom** [comment]
The OBIS network has been added to the DMNS Marine Invert resource on the VN IPT. I see this designation has been added to the dataset homepage at GBIF, too (https://www.gbif.org/dataset/5a6977c1-ece6-44b8-b71d-43cb1d4e0919). So far, I have not had to republish the DwC-A. Pending confirmation from @albenson-usgs that the data have been made available to OBIS, too. I can initiate a publishing event if necessary.

**Comment by albenson-usgs** [comment]
The DMNS Invertebrates collection is now in OBIS https://obis.org/dataset/4a606f86-2e7b-44a3-b50d-3d55a887a11f. Thanks everyone for the effort on this. In my mind the next steps are for DMNS to review what's there, document the steps in the process, and then determine if any other Arctos collections would like to share to OBIS.

**Comment by Jegelewicz** [comment]
@sharpphyl @acdoll can you document the process?

**Comment by sharpphyl** [comment]
This GitHub issue is a good start toward the documentation.  What exactly do you want @Jegelewicz ?  I can do a mini-tutorial video or just work with any other collection in Arctos that would like to contribute to OBIS.  I did a lot of review of specimen records and clean-up but Abby really did the majority of the work determining what records should be dropped. 

I've done a quick review of the OBIS link above and see no major issues, but I'll leave this issue open until I do a full review.

What other Arctos collections should we flag and encourage to participate?

**Comment by Jegelewicz** [comment]
@sharpphyl It would be great to just have a list of steps required. @lin-fred has stuff at NMMNH that would be good to add. Once Cal Poly Humboldt has their marine mammals in, they will definitely want to do this. I'll put it on their to-do list. BUT, we should be thinking about all collections that might have marine stuff. UAM:Mamm, all fish collections, ALMNH:Inv - @kmkocot for sure!

**Comment by Jegelewicz** [comment]
@dbloom YES! But especially whatever we need to do to let you know that we need you to take another terribly laborious yet adrenaline-filled action!

**Comment by dbloom** [comment]
OK @Jegelewicz

1. Identify the collection or collections for inclusion in OBIS.
2. A representative from Arctos, the identified collection, or OBIS (e.g., @albenson-usgs) should notify @dbloom via email to add said collection to the OBIS Network (notification via GitHub works, too, but email is more time efficient).
3. @dbloom will then prepare a poultice of green algae, nori and chocolate chips and apply it to the VertNet IPT in precise locations and in precise amounts. Getting the amount or location wrong could result, at best, in the assignment to the wrong network, and at worst, in massive data loss.
4. The IPT will be left in the sun for three days, over which @dbloom and members of the collection staff will fast and recite the complete Darwin Core Standard once every two hours until sunrise on the third day. OBIS will need to supply enough lemon water and vegetable broth to keep the recitation team nourished and awake. During the winter months, a tanning bed may be used in place of the sun.
5. At the end of the three days, the dried poultice will be removed and the residual will be burned in a cleaning fire of willow wood, sage and thyme.
6. @dbloom will update the VertNet IPT to include the OBIS Network to the resource metadata (an Administrative task).
7. @dbloom will notify Arctos, the collection and OBIS that the OBIS network has been added to the collections resource on the VN IPT.

Then we rest.

**Comment by albenson-usgs** [comment]
Want to add that those are basically all the steps that are needed except I would add step 2 to be: 
2. Perform a lookup to the World Register of Marine Species to check for updated marine taxonomy and update as appropriate. (Abby Benson can assist with this step)

**Comment by sharpphyl** [comment]
There's no way I can top David's description but I'll try to create a prosaic version of the steps I took to confirm Abby's lists and updates were the same as what I could have created.  If the other collections have questions during the process, I'll be glad to help.

**Comment by sharpphyl** [comment]
The following are the steps that I took to prepare our data for inclusion in OBIS.  I can't emphasize enough that the heavy lifting was done by Abby Benson(@albenson-usgs) and David Bloom (@dbloom).  

My role entailed getting our data cleaned up - which needed to be done anyway - and responding to specific questions Abby had.  The clean-up included:
1.  Checking all scientific names against WoRMS and updating invalid names.  Abby prepared a list of names with no match in WoRMS or that were invalid.  Most of those with no match were terrestrial species but all invalid names were updated.
2. Geolocating where possible as records without decimalLatitude and decimalLongitude are excluded.
3. Reviewing Abby's list of non-marine lots for reasonableness.  WoRMS associates all taxon names with its Environment so only those with Marine or Brackish environments are included. 
4. Other cleanup such as adding Individual Count to all records.

The first upload shows how many records were dropped and for what reason and lists missing fields.  This data will help us continue to improve the data quality.  Just over 60% of our total records met the OBIS standards. 

I do have one question for Abby.  In the list of dropped records (938) are those where WoRMS does not have a single exact match (ambiguous match).  Is there any way to overcome this as my preliminary check of these names shows that most are valid?   

I would enourage @lin-fred, @kmkocot and any others with marine specimens to consider adding your data to OBIS by commenting here to @albenson-usgs or in a new GitHub issue.  Let me know if you have any further questions.  Thanks again to Abby and David.

**Comment by Jegelewicz** [comment]
Thanks everyone for the input - I have assigned myself to write it up for the handbook.

**Comment by albenson-usgs** [comment]
> I do have one question for Abby. In the list of dropped records (938) are those where WoRMS does not have a single exact match (ambiguous match). Is there any way to overcome this as my preliminary check of these names shows that most are valid?

The species are [these ones](https://obis.org/table/nomatch?datasetid=4a606f86-2e7b-44a3-b50d-3d55a887a11f). Yes those do seem to be the ambiguous matches. The way to overcome this is to provide the LSID (e.g. urn:lsid:marinespecies.org:taxname:215457) in `scientificNameID`. OBIS actually prefers to have the LSID added to `scientificNameID` (i.e. it's a required term in OBIS) but we have relaxed that requirement when we are harvesting from an IPT that shares to GBIF and adds OBIS as a network to their datasets.

**Comment by sharpphyl** [comment]
By way of clarification, in some records, the aphiaID appears in our records in OBIS.
<img width="416" alt="Screen Shot 2022-06-18 at 2 35 10 PM" src="https://user-images.githubusercontent.com/15368365/174456303-ab7faaed-a961-4247-a428-2a8baa7583ac.png">

Is this coming from WoRMS and not from the Arctos records?  If so, the only records that would need the scientificNameID added would be those in the Taxa Not Matched list.

@dustymc @Jegelewicz Can scientificNameID be added as a field for the collections wanting to contribute to OBIS so that those catalog records are not excluded?  Would we have to manually add the WoRMS aphiaID to the record or could it be picked up from the taxonomic table?

**Comment by sharpphyl** [comment]
@albenson-usgs  Today when I opened our dataset page (https://obis.org/dataset/4a606f86-2e7b-44a3-b50d-3d55a887a11f), the details about the dataset have disappeared and it reads "Metadata only Dataset."  Has there been a change in what OBIS is pulling from VertNet?

**Comment by albenson-usgs** [comment]
@sharpphyl @dbloom the issue is on the VertNet IPT end. They have 0 records for your most recent publication of the data:
![Capture](https://user-images.githubusercontent.com/7586150/184900188-50c5fdca-997e-4e1a-a620-bb4f7a582308.PNG)

**Comment by Jegelewicz** [comment]
@dustymc @dbloom what's up?

**Comment by dbloom** [comment]
@Jegelewicz @dustymc @albenson-usgs I think this has happened before, but with a different dataset.

As far as I can tell from my end, the IPT is functioning properly and without error. Unless the collectionID has changed....

Using statement:  Select * from ipt_cache.occurrence where collectionID = 74

From here, it appears that the table on the Arctos side is not being populated.

**Comment by sharpphyl** [comment]
Thanks, Dusty.

**Comment by dbloom** [comment]
You know we love you @dustymc 

Some of us just have a strange way of showing it.

**Comment by Jegelewicz** [comment]
I love a good bromance

**Comment by dbloom** [comment]
Checked on this a few minutes ago and the IPT is still not seeing any records in the occurrence table. Has the refresh completed?

**Comment by dustymc** [comment]
no, it died, rebuilt everything and started it back up a few minutes ago

**Comment by dustymc** [comment]
And its done, confirmed to have done something. Do we have a staff exorcist with whom I can consult?

**Comment by dbloom** [comment]
Thank you @dustymc. Confirmed, I see records now. BUT.... An error:

Publishing version #20.61 of resource dmns_inv failed: Archive generation for resource dmns_inv failed: Can't validate DwC-A for resource dmns_inv. Each row in the occurrence file(s) must have a basisOfRecord, and each basisOfRecord must match the Darwin Core Type Vocabulary (please note comparisons are case insensitive)

Log Messages
Archive generation started for version #20.61 16:37:54
Start writing data file for Darwin Core Occurrence 16:37:54
Data file written for Darwin Core Occurrence with 31308 records and 95 columns 16:38:05
Start writing data file for Audubon Media Description 16:38:05
Data file written for Audubon Media Description with 0 records and 17 columns 16:38:05
All data files completed 16:38:05
EML file added 16:38:05
meta.xml archive descriptor written 16:38:05
Validating the core file: occurrence.txt. Depending on the number of records, this can take a while. 16:38:05
? Validating the core basisOfRecord is always present and its value matches the Darwin Core Type Vocabulary. 16:38:05
? Validating the core ID field occurrenceID is always present and unique. Note: the core ID field is required to link core records and extension records together. 16:38:05
✓ Validated each line has a occurrenceID, and each occurrenceID is unique 16:38:07
1 line(s) are missing a basisOfRecord 16:38:07
Archive validation failed, because not every row in the occurrence file(s) has a valid basisOfRecord (please note all basisOfRecord must match Darwin Core Type Vocabulary, and comparisons are case insensitive) 16:38:07
Restored version #20.60 of resource dmns_inv after publishing failure 16:38:08

**Comment by dbloom** [comment]
@dustymc I will add that sometimes if there is an empty row at the end of a table or spreadsheet, the IPT will read that as a occurrence without a basisOfRecord. I would be surprised if Arctos adds extra rows, but I thought it worth mentioning.

**Comment by dustymc** [comment]
See https://github.com/ArctosDB/arctos/issues/2432

**Comment by dbloom** [comment]
Ugh! Sorry @dustymc.

In this case with this collection, is it just that single record, or is DMNS:Inv:36482 just one example of many? Wondering if it is just the one, if it can be removed from the occurrence table so I can publish the dataset, after which I can turn off the auto-publishing until a solution has been found internally. That way your records are out there, even if they are not being updated.

Or, could you provide me with a download of the table with all of the data and I can make the removal and publish from a csv temporarily.

Just thinking out loud.

As an aside, I'm not sure that the new GBIF data model will use or recognize an expanded set of terms for basisOfRecord. That might be worth an inquiry so that you can know that whatever you decide in the short-term will not pop up again in a year.

Let me know if there is anything I can do.

**Comment by dustymc** [comment]
I did this:

```
arctosprod@arctos>> delete from ipt_cache.occurrence where  basisOfRecord is null;
DELETE 9111
```
and I'll rebuild the scripts to use it.

**Comment by dbloom** [comment]
+1  Let me know when it's up and I'll try to publish again.

**Comment by dustymc** [comment]
its up - scripts will just keep it there

**Comment by dbloom** [comment]
@dustymc That did it. Thank you.

@Jegelewicz @albenson-usgs This dataset is back up.

**Comment by sharpphyl** [comment]
My thanks to all of you!

**Comment by dbloom** [comment]
@sharpphyl All beer deliveries and other forms of homage go to @dustymc

**Comment by Jegelewicz** [comment]
Documentation started [here](https://handbook.arctosdb.org/how_to/How-to-Publish-to-OBIS.html). @sharpphyl @dustymc please let me know if anything needs to be added/edited or just change it yourself!

**Comment by dbloom** [comment]
Is this document the place where it should be noted that if Arctos collections are to be added/published to OBIS, I would need to be notified so that I can add the specific collection to the OBIS Network within the IPT? I think if that step doesn't happen, the data will be added the data won't get to OBIS.

**Comment by Jegelewicz** [comment]
Added above to the How-To

**Comment by sharpphyl** [comment]
Looks good.  I was going to add a link to Abby's GitHub page https://github.com/albenson-usgs but when I clicked on edit directly [here](https://github.com/ArctosDB/documentation-wiki/blob/gh-pages/_how_to/How-to-Publish_to_OBIS.markdown), this is what I got.  



<img width="1358" alt="Screen Shot 2022-08-30 at 6 37 55 AM" src="https://user-images.githubusercontent.com/15368365/187438462-6ac8b62d-b20a-4494-9523-dd7fad8b7ada.png">

**Comment by Jegelewicz** [comment]
@sharpphyl my bad! I fixed the typo - try it now.

**Comment by sharpphyl** [comment]
Works great, and a great idea for spreading the effort to update documentation.  I think I made and committed the suggested change correctly.

**Comment by Jegelewicz** [comment]
YAY! I took your good idea and added a link for Dave too.

**Comment by Jegelewicz** [comment]
I think we are done here.

**Comment by albenson-usgs** [comment]
I'm going to need help from @pieterprovoost on this. I think the OBIS processing went through some changes recently but I don't know the details and I'm not sure where to find them.

**Comment by dustymc** [comment]
@albenson-usgs question: Would is be possible for OBIS to deal with data like https://github.com/ArctosDB/arctos/issues/6532#issuecomment-1641119452?

In that case (one identification which references one WoRMS taxon) there's one ``"term":"urn:lsid:marinespecies.org:taxname:857376"``. Other records (although perhaps nothing that will be wanting to participate in OBIS) will have no such term, or 20 of them for an identification, or 20 of them for each of 20 identifications, or ....

**Comment by dustymc** [comment]
merge-->https://github.com/ArctosDB/arctos/issues/6532

**Comment by albenson-usgs** [comment]
@dustymc sorry I never responded to your question above. I lost track of it. I'm also unclear what you mean and obviously things have progressed since then so perhaps it's no longer relevant. Happy to get on a call so we can work through anything that's needed.

**Comment by dustymc** [comment]
Thanks @albenson-usgs - yes, we hope we have (or nearly have) drug what's needed out of the weeds!

**Comment by dbloom** [comment]
I might also be able to help @dustymc. Regardless of what you do manage to drag from the weeds, please be sure to provide me with some sort of notification that a given dataset should be published to OBIS so that I can update the IPT Network section appropriately.

**Comment by dustymc** [comment]
Thanks @dbloom - I'm not sure what's going on that far in, but I will be adding scientificNameID to the DWC data - let me know if that's going to break something and I'll do whatever you tell me to do.
